ts/api/pagopa.ts

```diff
@@ -0,0 +1,178 @@
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import { ApiFetchResult } from ".";
+import { Amount } from "../../definitions/pagopa/Amount";
+import { CreditCard as PagoPACreditCard } from "../../definitions/pagopa/CreditCard";
+import { Transaction as PagoPATransaction } from "../../definitions/pagopa/Transaction";
+import { Wallet } from "../../definitions/pagopa/Wallet";
+import { pagoPaApiUrlPrefix } from "../config";
+import I18n from "../i18n";
+import {
+  CreditCard,
+  CreditCardId,
+  CreditCardType,
+  UNKNOWN_CARD
+} from "../types/CreditCard";
+import { UNKNOWN_TRANSACTION, WalletTransaction } from "../types/wallet";
+
+/**
+ * This function converts a Wallet object into
+ * a displayable CreditCard one
+ * @param pWallet Wallet received from the PagoPA API
+ */
+const mapToLocalCreditCard = (pWallet: Wallet): CreditCard => {
+  if (!pWallet.creditCard) {
+    return UNKNOWN_CARD;
+  }
+  const pCard: PagoPACreditCard = pWallet.creditCard;
+  if (
+    pWallet.lastUsage === undefined ||
+    pWallet.idWallet === undefined ||
+    pCard.pan === undefined ||
+    pCard.holder === undefined ||
+    pCard.expireMonth === undefined ||
+    pCard.expireYear === undefined ||
+    pCard.brandLogo === undefined ||
+    CreditCardType.decode(pCard.brandLogo).isLeft()
```

Comment:
you can use `!CreditCardType.is(pCard.brandLogo)`

Reply:
btw it looks like all these properties are mandatory in the CreditCard type below, so I guess `CreditCard.is(pCard)` should suffice (or `CreditCard.decode(pCard).isLeft()`)